### PR TITLE
Add support for uploading to S3 directories

### PR DESCRIPTION
**Pull request by pcantea**
Allows for specifying s3 buckets in the format `s3://bucket/key` or `bucket/path/to/key`

### CODE REVIEW

File: pkg/popeye.go [hunk 1]
```diff
@@ -34,6 +36,8 @@ var (
 	LogFile = filepath.Join(os.TempDir(), "popeye.log")
 	// DumpDir indicates a directory location for sanitizer reports.
 	DumpDir = dumpDir()
+	// ErrUnknownS3BucketProtocol defines the error if we can't parse the S3 URI
+	ErrUnknownS3BucketProtocol = errors.New("invalid S3 URI: hostname not valid")
```
Comment (derailed) [inline pkg/popeye.go:40]: Is this sentinel error necessary? Don't think there is much reuse for it as it is defined in a very specific context.

File: pkg/popeye.go [hunk 2]
```diff
@@ -478,6 +486,37 @@ func dumpDir() string {
 	return filepath.Join(os.TempDir(), "popeye")
 }
 
+func parseBucket(bucketURI string) (string, string, error) {
```
Comment (derailed) [inline pkg/popeye.go:489]: Should probably add a test for this function to make sure the parsing produces what we want.

File: pkg/popeye.go [hunk 3]
```diff
@@ -478,6 +486,37 @@ func dumpDir() string {
...
+	case "":
+		uri := strings.SplitAfterN(strings.Trim(bucketURI, "/"), "/", 2)
+		bucket = strings.Trim(uri[0], "/")
+		if len(uri) == 1 {
+			key = ""
+		} else {
+			key = uri[1]
+		}
+	default:
+		return "", "", ErrUnknownS3BucketProtocol
```
Comment (derailed) [inline pkg/popeye.go:514]: Might as well return a canned error here vs the sentinel error??

File: pkg/popeye.go (hunk 1 above)
Comment (pcantea) [reply pkg/popeye.go:40]: I kept the sentinel error to help with testing, would that be ok or would you rather me still remove it?

**Comment by derailed** [comment]
@pcantea Thank you for making these corrections! Looks good! Could you update the README as well to make sure we're documenting this change? Thank you!!

**Pull request merged**